### PR TITLE
feat(Authoring): Implement Peer Group Activity Tag authoring

### LINKED ISSUE #479: Implement Peer Group Activity Tag authoring
Implement it for all components that save student work.

### PR DESCRIPTION
**Pull request by geoffreykwan**
Make sure you can author the Peer Group Activity Tag in the advanced authoring view for components that save work.

Closes #479

### CODE REVIEW

File: src/app/authoring-tool/edit-component-peer-group-activity-tag/edit-component-peer-group-activity-tag.component.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    const tagsFound = new Map();
+    for (const component of this.projectService.getComponents()) {
+      const tag = component.peerGroupActivityTag;
+      if (tag != null && tag !== '' && !tagsFound.has(tag)) {
+        tags.push(tag);
+        tagsFound.set(tag, true);
+      }
+    }
+    return tags.sort();
+  }
```
Comment (hirokiterashima) [inline src/app/authoring-tool/edit-component-peer-group-activity-tag/edit-component-peer-group-activity-tag.component.ts]: Would it be cleaner if tagsFound is a Set<string> in this function? 
```suggestion
  getPeerGroupActivityTags(): string[] {
    const tagsFound = new Set();
    for (const component of this.projectService.getComponents()) {
      const tag = component.peerGroupActivityTag;
      if (tag != null && tag !== '') {
        tagsFound.add(tag);
      }
    }
    return Array.from(tagsFound).sort();
  }
```

File: src/assets/wise5/components/peerChat/edit-peer-chat-advanced-component/edit-peer-chat-advanced-component.component.ts [hunk 2]
```diff
@@ -20,5 +20,7 @@ export class EditPeerChatAdvancedComponentComponent extends EditAdvancedComponen
     super(NodeService, NotebookService, ProjectService);
   }
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    super.ngOnInit();
+  }
```
Comment (hirokiterashima) [inline src/assets/wise5/components/peerChat/edit-peer-chat-advanced-component/edit-peer-chat-advanced-component.component.ts]: Not necessary?

```suggestion
```

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Each time I type a letter in the peerGroupActivityTag input box, it creates a new row in the peer_group_activities table because it's saving changes per keystroke. Can you change this to save when input loses focus, or after a debounce?

Also, see code comments inline.

**Comment by geoffreykwan** [comment]
I wonder if losing focus or debounce won't be good enough. Both of those can still lead to unwanted rows in the database.

Should we have a confirm dialog when adding a new tag?
Should we check for duplicate tags before saving a tag?
Should authors be able to delete tags?

**Review by hirokiterashima — APPROVED** [review]
Looks good.

**Pull request merged**